From 416ba77bd083496c5a1bc076ef5e95480d8cff79 Mon Sep 17 00:00:00 2001
From: Zahari Doychev <zdoychev@maxlinear.com>
Date: Thu, 18 Mar 2021 19:02:27 +0100
Subject: [PATCH] dpm: pon_qos: check bands range only in replace

PONRTSYS-8730: The tc prio bands needs to be checked only in the replace
command as in case of destroy/dump/stats the bands can have invalid value.
---
 drivers/net/datapath/pon_qos/pon_qos_tc_prio.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_prio.c b/drivers/net/datapath/pon_qos/pon_qos_tc_prio.c
index 116b2db69116..d5b8da7217aa 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_prio.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_prio.c
@@ -87,7 +87,7 @@ static int pon_qos_tc_prio_replace(struct net_device *dev,
 	if (!port) {
 		port = pon_qos_port_alloc(dev);
 		if (!port) {
-			netdev_err(dev, "tc-drr port alloc failed\n");
+			netdev_err(dev, "tc-prio port alloc failed\n");
 			return -ENOMEM;
 		}
 		newp = true;
@@ -180,13 +180,13 @@ int pon_qos_tc_prio_offload(struct net_device *dev,
 
 	ASSERT_RTNL();
 	netdev_dbg(dev, "PRIO: offload starting\n");
-	if (opt->replace_params.bands > PON_QOS_TC_MAX_Q) {
-		netdev_dbg(dev, "tc prio offload - out of range bands %d\n",
-			   opt->replace_params.bands);
-		return -EINVAL;
-	}
 	switch (opt->command) {
 	case TC_PRIO_REPLACE:
+		if (opt->replace_params.bands > PON_QOS_TC_MAX_Q) {
+			netdev_err(dev, "tc-prio - out of range bands %d\n",
+			   opt->replace_params.bands);
+			return -EINVAL;
+		}
 		netdev_dbg(dev, "replace pid:%#x class/handle:%#x bands:%#x\n",
 			   opt->parent, opt->handle, opt->replace_params.bands);
 		err = pon_qos_tc_prio_replace(dev, handle, opt,
